diff --git a/src/root_port_capture/headless_capture.cpp b/src/root_port_capture/headless_capture.cpp
--- a/src/root_port_capture/headless_capture.cpp
+++ b/src/root_port_capture/headless_capture.cpp
@@ -13,6 +13,7 @@
 #include "core/core.h"
 #include "core/core_timing.h"
 #include "core/frontend/emu_window.h"
+#include "core/frontend/applets/default_applets.h"
 #include "core/loader/loader.h"
 #include "core/hle/service/service.h"
 #include "root_port_capture/gpu_command_capture.h"
@@ -116,6 +117,7 @@
         }
     }
     auto& system = Core::System::GetInstance();
+    Frontend::RegisterDefaultApplets(system);
     auto& movie = system.Movie();
     if (argument_count >= 4) {
         movie.PrepareForPlayback(std::filesystem::absolute(arguments[3]).string());
