diff --git a/src/core/hle/service/fs/file.cpp b/src/core/hle/service/fs/file.cpp
index b37c19d..9cc6325 100644
--- a/src/core/hle/service/fs/file.cpp
+++ b/src/core/hle/service/fs/file.cpp
@@ -140,7 +140,8 @@ void File::Read(Kernel::HLERequestContext& ctx) {
             }
 
             const auto read_delay = static_cast<s64>(backend->GetReadDelayNs(async_data->length));
-            if (!async_data->cache_ready) {
+            if (!async_data->cache_ready &&
+                !Settings::values.deterministic_async_operations.GetValue()) {
                 const auto time_took = std::chrono::duration_cast<std::chrono::nanoseconds>(
                                            std::chrono::steady_clock::now() - async_data->pre_timer)
                                            .count();
