--- a/src/core/file_sys/archive_extsavedata.cpp
+++ b/src/core/file_sys/archive_extsavedata.cpp
@@ -43,7 +43,7 @@
         if (offset > size) {
             return ResultWriteBeyondEnd;
         } else if (offset == size) {
-            return 0ULL;
+            return std::size_t{0};
         }
 
         if (offset + length > size) {
--- a/src/core/file_sys/archive_ncch.cpp
+++ b/src/core/file_sys/archive_ncch.cpp
@@ -294,7 +294,7 @@
                                        const bool update_timestamp, const u8* buffer) {
     LOG_ERROR(Service_FS, "Attempted to write to NCCH file");
     // TODO(shinyquagsire23): Find error code
-    return 0ULL;
+    return std::size_t{0};
 }
 
 u64 NCCHFile::GetSize() const {
--- a/src/core/file_sys/ivfc_archive.cpp
+++ b/src/core/file_sys/ivfc_archive.cpp
@@ -105,7 +105,7 @@
                                        const bool update_timestamp, const u8* buffer) {
     LOG_ERROR(Service_FS, "Attempted to write to IVFC file");
     // TODO(Subv): Find error code
-    return 0ULL;
+    return std::size_t{0};
 }
 
 u64 IVFCFile::GetSize() const {
@@ -137,7 +137,7 @@
                                                const u8* buffer) {
     LOG_ERROR(Service_FS, "Attempted to write to IVFC file");
     // TODO(Subv): Find error code
-    return 0ULL;
+    return std::size_t{0};
 }
 
 u64 IVFCFileInMemory::GetSize() const {
